builtin/providers/google/resource_dataflow_job_test.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataflowJobCreate(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDataflowJob,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(
+						"google_dataflow_job.big_data"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDataflowJobDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_dataflow_job" {
+			continue
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		job, _ := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if job != nil {
+			return fmt.Errorf("Job still present")
+		}
+	}
+
+	return nil
+}
+
+func testAccDataflowJobExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Job does not exist")
+		}
+
+		return nil
+	}
+}
+
+var testAccDataflowJob = fmt.Sprintf(`
```

Comment:
Thank you! That clears things up.

So I'm not sure it actually matters whether we use the name or id as the id. If we add logic into the read function that says to remove the config from state if the job is stopped, then the next `apply` will try to recreate the job. Unless I'm missing something, it shouldn't behave any differently if it's the name or the id. The big thing where the `Id` field comes into play is with `import`, since you generally import by Id. It doesn't look like you've added import support yet, but I could see it getting added in the future. So I guess the question becomes, would we prefer users to import with `name` or `id`? I think users would probably want `name`? But it also becomes more difficult code-wise since the Dataflow API expects the Id everywhere (so for import, if we only have the name, we can't just call the API and get the job- we'd have to list all jobs and then find the one with the matching name. Doable, but annoying). What do you think?